Assets/Scripts/PlayerMovement.cs

```diff
@@ -85,6 +87,19 @@ public class PlayerMovement : MonoBehaviour
         if (action != null) action.execute();
         else Debug.Log("Item has no action");
     }
+    void handleEat() {
+        var healthFill = currentlyGrabbed.GetComponent<ItemAssociation>().item.healthFill;
+        if (healthFill > 0) {
+            healthUI.SetHealth(healthUI.actualHealth + healthFill);
+            Destroy(currentlyGrabbed);
+        } else {
+            Debug.Log("You can't eat that");
+        }
+    }
+
+    public void TakeDamage(float damage) {
+        healthUI.SetHealth(healthUI.actualHealth - damage);
```

Comment:
I would assign the responsibility of keeping track of the health value in the player object. It is after all the player who takes and heals damage, the UI is only there to display it.